lib/basedriver/driver.js

```diff
@@ -188,6 +188,16 @@ class BaseDriver extends MobileJsonWireProtocol {
     return true;
   }
 
+  /**
+   * Test createSession inputs to see if this is a W3C Session or a MJSONWP Session
+   */
+  determineProtocol (desiredCapabilities, requiredCapabilities, capabilities) {
+    if (_.isObject(capabilities) && _.isObject(capabilities.alwaysMatch)) {
+      return this.protocol = 'W3C';
+    }
+    this.protocol = 'MJSONWP';
```

Comment:
`return (_.isPlainObject(capabilities) && _.isPlainObject(capabilities.alwaysMatch)) ? W3C : MJSONWP`

Reply:
I would agree that this would be better as a pure function, that just returns the correct protocol. No longer needs to be called in the context of the driver (can be a utility method instead).